### PR TITLE
UI/UX fixes: Reduce 3d viewer loading time + fix blurry renders on high-end displays

### PR DESCRIPTION
**Pull request by fulopkovacs**
# TLDR

First of all: I love the project, thank you for your work! ☺️ 

After @gregzaal's talk at Blender Conf 2022, I realized that the source code is open, so I decided to make a PR that addresses some bugs I found when I first interacted with the site. I'm talking about these ones:

1. **UI:** 3D viewer loads too slow (cause: textures are too big)
2. **UX:** no response to the user while the 3D viewer is loading (at first I thought it’s not loading at all)
3. **UI:** blurry renders on devices with high device pixel ratio (e.g.: macBook, iPhone), despite the 4k textures

I can open up separate issues/PR-s for them if you want.

# Detailed explanation

## Issue 1: [UI] 3D viewer loads too slow (cause: textures are too big)

- How slow? On a M1 macBook, 50 MB/s internet:

  | ORIGINAL (4k textures) | This PR (2k textures)
-- | -- | --
loading the 3D model in the viewer from start to finish | ~6.5s | ~3s
largest texture | 5s | 1.5s

- personally, I think this speed tradeoff isn't worth it, especially considering that the renders will be blurry anyways on macBooks/iPhones/high-end displays (see **Issue 3**)
- Check this video where I compare the performance profiles (made with Firefox) of the original version of the site and my PR:

https://user-images.githubusercontent.com/43729152/200120952-b7663cfa-7850-441f-80fd-6ece78c47bbb.mp4

  - gzipped profiles:
    - [polyhaven-profile-ORIGINAL.gz](https://github.com/Poly-Haven/polyhaven.com/files/9943854/polyhaven-profile-ORIGINAL.gz)
    - [polyhaven-profile-PR.gz](https://github.com/Poly-Haven/polyhaven.com/files/9943855/polyhaven-profile-PR.gz)

## **UX:** no response to the user while the 3D viewer is loading (at first I thought it was not loading at all)
- When I first used polyhaven I thought that the 3D viewer didn’t work, because nothing happened after I clicked on the “View in 3D”-button.
- In this PR I added a loading spinner that is present until the canvas starts rendering the model
  - this PR:
  
    https://user-images.githubusercontent.com/43729152/200121235-17bc672b-3b79-4b3a-a197-c380a3e2c233.mp4
  
  - original: 
  
    https://user-images.githubusercontent.com/43729152/200121232-189d1cf6-988d-4b58-b4b9-12766b9901c5.mp4
   
## Issue 3. [UI] Blurry textures on devices with higher `devicePixelRatio` (MacBook, iPhone, etc)
- if the `dpr` is misconfigured, 4k textures might look as blurry as lower res textures
- see the differences here (**fun fact:** the sharper renders use 2k textures (the blurrier ones have 4k textures)):

     https://user-images.githubusercontent.com/43729152/200121323-5432d862-9800-4b35-bd26-adc40d1d975d.mp4
     - this pr:
       dpr-NEW
     - original: 
      dpr-ORIGINAL
      
- **cause:** this project uses an older version of `r3f`, where the `[1,2]` is not the default for the `dpr` attribute of the `<Canvas />` component.

# Closing thoughts

I can break up this PR into smaller ones if you want, or create issues for them, but I figured it'd make the communication in one place. Feel free to ask me for clarifications on my changes too!

### CODE REVIEW

**Comment by gregzaal** [comment]
Hey, thanks for the PR! I should have mentioned in the readme that the `dev` branch is the one to work on - I've already fixed the loading delay/lack of spinner issue there.

If you can submit a separate PR for the `dpr` issue built against the `dev` branch that'd be good to go :)

As for switching the preview resolution down to 2K to save time, I'd rather not do this. 4K is a good compromise between loading time and preview quality (most of our assets are at least 8K if not more).
